libkmod/libkmod-elf.c

```diff
@@ -476,10 +476,12 @@ int kmod_elf_get_modversions(const struct kmod_elf *elf, struct kmod_modversion
 		struct kmod_modversion32 *mv;
 
 		crclen = sizeof(mv->crc);
+		namlen = sizeof(mv->name);
 	} else {
 		struct kmod_modversion64 *mv;
 
 		crclen = sizeof(mv->crc);
+		namlen = sizeof(mv->name);
```

Comment:
Instead of duplicating this even more, let's make it a helper function.

The `verlen` piece (seen below) can become a static assert (`assert_cc` in kmod), since 32/64bit versions of the struct are intentionally kept identical.

Reply:
I've moved it into its own inline function, keeping `assert_cc` and `verlen`. Since the function is inlined and compilers are smart enough, the verlen is a constant anyway, so the preprocessor definition can be removed instead.

Reply:
Speaking on inline: We should probably nuke the "Wno-inline" flag ... on a rainy day.